std/assembly/bindings/wasi_snapshot.ts

```diff
@@ -869,22 +869,26 @@ export namespace errno {
 export type errno = u16;
 
 /** An event that occurred. */
-@unmanaged export abstract class event {
+@unmanaged export class event {
   /** User-provided value that got attached to `subscription#userdata`. */
   userdata: userdata;
   /** If non-zero, an error that occurred while processing the subscription request. */
   error: errno;
-  /* The type of the event that occurred. */
+  /** The type of the event that occurred. */
   type: eventtype;
+  /** The contents of the event. */
+  u: event_fd_readwrite;
```

Comment:
Iirc this is a union, but `u` would point somewhere here. Idea previously was that after `type` there is `__padding0` followed by the extensions performed by `rwevent` or any other concrete event. One would then cast `event` to `rwevent` if `event#type` indicates an `rwevent`, in turn gaining access to the extended data of the union **if** an rwevent.

Reply:
Hmm, so I should revert this changes and use previous approach?

Reply:
It's fine to update the layout if it has changed ofc, but the new approach with `u` doesn't seem correct as in AS every field of a class type is a pointer, and there is no pointer here. The only way to model a union is with `extends` I think.